packages/nextjs/components/scaffold-eth/Contract/utilsContract.tsx

```diff
@@ -25,6 +23,13 @@ const getDeployedContract = (
     return;
   }
 
+  let ContractData;
+  try {
+    ContractData = require("~~/generated/hardhat_contracts.json");
+  } catch (e) {
+    return;
+  }
+
   const contractsAtChain = ContractData[chainId as keyof typeof ContractData];
```

Comment:
it will break here if ContractData isn't set

Reply:
But I think if the interpreter reaches this line then its sure that ContractData will be set, even Typescript does not complain but I might have missed something 🙌

Reply:
no, ts thinks it has `any` type

Reply:
Yeah, but I don't think it will be `undefined` but I am sure I am missing something

Reply:
So, what's the verdict here? :D

If the file doesn't exist an error will be thrown, catched and the function will return, so line 33 won't be executed. Right?

The idea is to avoid showing an error when the builder forgot to run "yarn deploy".

Reply:
My bad, I didn't see `return` in the catch block, and thought a function will continue it's execution. Sorry 🤷‍♂️